src/org/traccar/BaseEventHandler.java

```diff
@@ -12,24 +14,32 @@ public boolean isLastPosition() {
         return isLastPosition;
     }
 
+    private Device device;
```

Comment:
Why are you storing in the event handler. I didn't actually notice it before, but `isLastPosition` shouldn't be here either. One handler can be used for different devices.

By the way, there is another typo that I just noticed. It should be `analyze` instead of `analize`.